### ISSUE TITLE
Silently failing

### ISSUE THREAD

**Issue by retog** [coderaiser/node-jaguar#1]
I'm using jaguar in a web-application where I get tar.gz data via HTTP. I've noticed that when I accidentaly uplad the data as text rather than binary (using `--data` rather than `--data-binary` with cURL) none of the extraction events get triggerered. I would expect an error event to be sent.

This is my code:
```
return new Promise((resolve,reject) => {
        const extract = jaguar.extract(from, path.join(from+"-expanded"));
        extract.on('error', (error) => {
            reject(error)
        });
        extract.on('end', () => {
            resolve();
        });        
        extract.on('file', (name) => {
            console.log(name);
        });
        extract.on('progress', (percent) => {
            console.log(percent + '%');
        });
        console.log("Unzipping results promised. Will be at "+from+"-expanded");
    });
```

**Comment by coderaiser** [comment]
I can't reproduce this, what version do you use?
I tried with:

```js
const jaguar = require('.');
const {join} = require('path');

const fn = new Promise((resolve,reject) => {
    const from = './package.json';
    const extract = jaguar.extract(from, join(from + '-expanded'));
    extract.on('error', (error) => {
        reject(error);
    });
    extract.on('end', () => {
        resolve();
    });
    extract.on('file', (name) => {
        console.log(name);
    });
    extract.on('progress', (percent) => {
        console.log(percent + '%');
    });
    console.log('Unzipping results promised. Will be at ' + from + '-expanded');
});

fn.then(console.log).catch(console.error);
```

And here is what I saw:

```sh
coderaiser@cloudcmd:~/node-jaguar$ node test.js
Unzipping results promised. Will be at ./package.json-expanded
Error: Invalid tar header. Maybe the tar is corrupted or it needs to be gunzipped?
    at Object.exports.decode (/home/coderaiser/node-jaguar/node_modules/tar-stream/headers.js:267:43)
```

**Comment by retog** [comment]
I'm using version 5.0.1

Using your code I get the following output:
```
> jaguar-test@1.0.0 start /mnt/c/Users/me/Documents/jaguar-test
> node index.js

Unzipping results promised. Will be at ./package.json-expanded
Error: Unexpected end of data
    at Extract._final (/mnt/c/Users/me/Documents/jaguar-test/node_modules/tar-stream/extract.js:253:42)
    at callFinal (/mnt/c/Users/me/Documents/jaguar-test/node_modules/tar-stream/node_modules/readable-stream/lib/_stream_writable.js:586:10)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
which is OK.

But when instead of package.json I use 
[broken.tar.gz](https://github.com/coderaiser/node-jaguar/files/3617445/broken.tar.gz) I don't get an error message but only the following:
```
> jaguar-test@1.0.0 start /mnt/c/Users/me/Documents/jaguar-test
> node index.js

Unzipping results promised. Will be at ./broken.tar.gz-expanded
```
neither of the functions passed to `then` and `catch` is invoked.

The output with a valid archive is as follows:
```
> jaguar-test@1.0.0 start /mnt/c/Users/me/Documents/jaguar-test
> node index.js

Unzipping results promised. Will be at ./good.tar.gz-expanded
20%
rdf.ttl
40%
blobs/Un/
60%
blobs/Un/0tv_/
80%
blobs/Un/0tv_/82esSVoI/
100%
blobs/Un/0tv_/82esSVoI/Un0tv_82esSVoIn2fxO45
```

**Comment by coderaiser** [comment]
Now I can reproduce :). Fixed with https://github.com/coderaiser/pipe-io/commit/0c3d9833afcfcc2570d13f4a7d259e53ae16f59a, Landed in [pipe-io v3.0.12](https://github.com/coderaiser/pipe-io/releases/tag/v3.0.12).

Please reinstall `jaguar`.
Is it works for you?

**Comment by retog** [comment]
It works! Many thanks.
